server/channels/app/platform/web_conn.go

```diff
@@ -301,10 +311,26 @@ func (wc *WebConn) GetConnectionID() string {
 // SetActiveChannelID sets the active channel id of the connection.
 func (wc *WebConn) SetActiveChannelID(id string) {
 	wc.activeChannelID.Store(id)
+	if id != UnsetPresenceIndicator {
+		wc.setActiveOnce.Do(func() {
+			// We set these to empty as soon as we get an active channel scope.
+			// We do this to let the optimization take effect sooner, rather than
+			// waiting for a user to open a thread.
```

Comment:
Why not have the client decide/set this? I am slightly concerned by the use of implicit logic like this to essentially track client state.

Reply:
> Why not have the client decide/set this?

What are you thinking? 

The real problem is deciding between unset state and empty state to allow mobile clients to work.

Reply:
I am thinking that the client could set these values to empty when sending the active channel vs the server implicitly doing so.

Reply:
Oh then we have to differentiate between first load of webapp, vs channel switch. Since this is at the component level, the component doesn't really know what's happening on the outside. Maybe I can make the change when the ws client gets initialized. Will look into it.

Reply:
Cool. By the way, this more of a general thought than a request for changes. I am just worried that assuming client state from the server side could quickly become tricky and prone to errors, but for now there's nothing wrong with any of this.

Reply:
Yes agree. Will get rid of this.

Reply:
I've got rid of this completely now. The downside is that the optimization doesn't kick in until you open one thread atleast. But the upside is that the code is much clear, and the assumptions are explicit between client and server rather than anything implicit.